storage/src/mmr/bitmap.rs

```diff
@@ -31,108 +31,114 @@ impl<H: CHasher + Send + Sync> Storage<H::Digest> for BitmapStorage<'_, H> {
-/// overhead (e.g. by a factor of >256 for 32-byte digests).
-pub struct Bitmap<H: CHasher> {
-    /// The bitmap itself, in chunks of size DIGEST_SIZE bytes. The number of valid bits in the last
-    /// chunk is given by `self.next_bit`. Within each byte, lowest order bits are treated as coming
-    /// before higher order bits in the bit ordering.
+/// Merkelization of the bitmap is performed over chunks of N bytes. If the goal is to minimize
+/// proof sizes, choose an N that is equal to the size or double the size of the hasher's digest.
+pub struct Bitmap<H: CHasher, const N: usize> {
+    /// The bitmap itself, in chunks of size N bytes. The number of valid bits in the last chunk is
+    /// given by `self.next_bit`. Within each byte, lowest order bits are treated as coming before
+    /// higher order bits in the bit ordering.
     ///
     /// Invariant: The last chunk in the bitmap always has room for at least one more bit.
-    bitmap: Vec<u8>,
+    bitmap: VecDeque<[u8; N]>,
 
     /// The position within the last chunk of the bitmap where the next bit is to be appended.
     ///
-    /// Invariant: This value is always in the range [0, DIGEST_SIZE * 8).
+    /// Invariant: This value is always in the range [0, N * 8).
     next_bit: u64,
 
-    /// A Merkle tree with each leaf representing DIGEST_SIZE*8 bits of the bitmap.
+    /// A Merkle tree with each leaf representing N*8 bits of the bitmap.
     ///
-    /// When a chunk of DIGEST_SIZE*8 bits is accumulated by the bitmap, it is added to this tree.
-    /// Because leaf elements can be updated when bits in the bitmap are flipped, this tree, while
-    /// based on an MMR structure, is not an MMR but a Merkle tree.  The MMR structure results in
-    /// reduced update overhead for elements being appended or updated near the tip compared to a
-    /// more typical balanced Merkle tree.
+    /// When a chunk of N*8 bits is accumulated by the bitmap, it is added to this tree. Because
+    /// leaf elements can be updated when bits in the bitmap are flipped, this tree, while based on
+    /// an MMR structure, is not an MMR but a Merkle tree.  The MMR structure results in reduced
+    /// update overhead for elements being appended or updated near the tip compared to a more
+    /// typical balanced Merkle tree.
     mmr: Mmr<H>,
 }
 
-impl<H: CHasher> Default for Bitmap<H> {
+impl<H: CHasher, const N: usize> Default for Bitmap<H, N> {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl<H: CHasher> Bitmap<H> {
+impl<H: CHasher, const N: usize> Bitmap<H, N> {
     /// The size of a chunk in bytes.
-    pub const CHUNK_SIZE: usize = H::Digest::SIZE;
+    pub const CHUNK_SIZE: usize = N;
 
     /// The size of a chunk in bits.
-    const CHUNK_SIZE_BITS: u64 = Self::CHUNK_SIZE as u64 * 8;
+    const CHUNK_SIZE_BITS: u64 = N as u64 * 8;
 
     /// Return a new empty bitmap.
     pub fn new() -> Self {
+        let bitmap = VecDeque::from([[0u8; N]]);
```

Comment:
nit: May be better to add a `capacity` arg to `new()` and then use `with_capacity` to create and then `push([0u8; N])`

Reply:
We can skip this for now.